src/librustc/infer/error_reporting/anon_anon_conflict.rs

```diff
@@ -54,30 +54,57 @@ impl<'a, 'gcx, 'tcx> InferCtxt<'a, 'gcx, 'tcx> {
         let scope_def_id_sub = anon_reg_sub.def_id;
         let bregion_sub = anon_reg_sub.boundregion;
 
-        let ty_sup = or_false!(self.find_anon_type(sup, &bregion_sup));
+        let anon_ty_sup = or_false!(self.find_anon_type(sup, &bregion_sup));
 
-        let ty_sub = or_false!(self.find_anon_type(sub, &bregion_sub));
+        let anon_ty_sub = or_false!(self.find_anon_type(sub, &bregion_sub));
+        let (ty_sup, ty_fndecl_sup) = anon_ty_sup;
+        let (ty_sub, ty_fndecl_sub) = anon_ty_sub;
 
-        let (main_label, label1, label2) = if let (Some(sup_arg), Some(sub_arg)) =
+        let (spanfin1, spanfin2, main_label, label) = if let (Some(sup_arg),
+                                                                       Some(sub_arg)) =
             (self.find_arg_with_anonymous_region(sup, sup),
              self.find_arg_with_anonymous_region(sub, sub)) {
 
-            let (anon_arg_sup, is_first_sup, anon_arg_sub, is_first_sub) =
-                (sup_arg.arg, sup_arg.is_first, sub_arg.arg, sub_arg.is_first);
-            if self.is_self_anon(is_first_sup, scope_def_id_sup) ||
-               self.is_self_anon(is_first_sub, scope_def_id_sub) {
-                return false;
-            }
-
-            if self.is_return_type_anon(scope_def_id_sup, bregion_sup) ||
-               self.is_return_type_anon(scope_def_id_sub, bregion_sub) {
-                return false;
-            }
+            let (anon_arg_sup, anon_arg_sub) = (sup_arg.arg, sub_arg.arg);
```

Comment:
If we do the `or_false!` transformation above, and we don't care about the other field, we could do:

```
let AnonymousArgInfo { arg: anon_arg_sup, .. } = or_false!(self.find_arg_with_anonymous_region(sup, sup));
```

Not sure if this is a net win.

Reply:
I'm a bit confused about what's happening here